packages/aws-cdk-lib/aws-cloudfront-origins/lib/s3-bucket-origin.ts

```diff
@@ -0,0 +1,251 @@
+        this.originAccessIdentity = props?.originAccessIdentity;
+      }
+
+      public bind(scope: Construct, options: cloudfront.OriginBindOptions): cloudfront.OriginBindConfig {
+        if (!this.originAccessIdentity) {
+          // Using a bucket from another stack creates a cyclic reference with
+          // the bucket taking a dependency on the generated S3CanonicalUserId for the grant principal,
+          // and the distribution having a dependency on the bucket's domain name.
+          // Fix this by parenting the OAI in the bucket's stack when cross-stack usage is detected.
+          const bucketStack = Stack.of(bucket);
+          const bucketInDifferentStack = bucketStack !== Stack.of(scope);
+          const oaiScope = bucketInDifferentStack ? bucketStack : scope;
+          const oaiId = bucketInDifferentStack ? `${Names.uniqueId(scope)}S3Origin` : 'S3Origin';
+
+          this.originAccessIdentity = new cloudfront.OriginAccessIdentity(oaiScope, oaiId, {
+            comment: `Identity for ${options.originId}`,
+          });
+        };
+        // Used rather than `grantRead` because `grantRead` will grant overly-permissive policies.
+        // Only GetObject is needed to retrieve objects for the distribution.
+        // This also excludes KMS permissions; OAI only supports SSE-S3 for buckets.
+        // Source: https://aws.amazon.com/blogs/networking-and-content-delivery/serving-sse-kms-encrypted-content-from-s3-using-cloudfront/
+        const result = bucket.addToResourcePolicy(new iam.PolicyStatement({
+          resources: [bucket.arnForObjects('*')],
+          actions: ['s3:GetObject'],
+          principals: [this.originAccessIdentity.grantPrincipal],
+        }));
+        if (!result.statementAdded) {
+          Annotations.of(scope).addWarningV2('@aws-cdk/aws-cloudfront-origins:updateImportedBucketPolicyOai',
+            'Cannot update bucket policy of an imported bucket. You will need to update the policy manually instead.\n' +
+            'See the "Setting up OAI with imported S3 buckets (legacy)" section of module\'s README for more info.');
+        }
+        return this._bind(scope, options);
+      }
+
+      protected renderS3OriginConfig(): cloudfront.CfnDistribution.S3OriginConfigProperty | undefined {
+        if (!this.originAccessIdentity) {
+          throw new Error('Origin access identity cannot be undefined');
+        }
+        return { originAccessIdentity: `origin-access-identity/cloudfront/${this.originAccessIdentity.originAccessIdentityId}` };
+      }
+    }();
+  }
+
+  /**
+   * Create a S3 Origin with default S3 bucket settings (no origin access control)
+   */
+  public static withBucketDefaults(bucket: IBucket, props?: cloudfront.OriginProps): cloudfront.IOrigin {
+    return new class extends S3BucketOrigin {
+      constructor() {
+        super(bucket, { ...props });
+      }
+    }();
+  }
+
+  constructor(bucket: IBucket, props: S3BucketOriginBaseProps) {
```

Comment:
should the `prop` be an optional? It seems that all properties are optional and `withOriginAccessControl` can take undefined props.

Reply:
yes good catch, will update